### ISSUE TITLE
Unable to map error to Linux FUSE

### ISSUE THREAD

**Issue by tronsoft** [docker/for-win#5518]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: D6A81DFA-A70C-40E1-946F-7CFF55EFFAFB/20200121201853

### Expected behavior
Run `dotnet test` with an error
### Actual behavior
MsTestAdapter v2 fails
### Information

  - Windows Version: 17763.864
  - Docker Desktop Version: 2.2.0.42247
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: No

### Steps to reproduce the behavior

  1. run `docker run -it -v C:\Users\...\TestCreateDirectory\:/mnt  --rm --name testit mcr.microsoft.com/dotnet/core/sdk:3.1`
  2. in the container goto /mnt
  3. run: `dotnet test <name of a unittest project>.dll`
  4. The mstestadapter:v2 has an i/o error

In the log file the following error is shown:
[20:38:13.368][GoBackendProcess  ][Error  ] time="2020-01-21T20:38:13+01:00" msg="Unable to map error to Linux FUSE errno: mkdir D:\\dev\\adir\\anotherdir\\bin\\Debug\\netcoreapp3.1\\TestResults\\Deploy_ 2020-01-21 19:38:13: The filename, directory name, or volume label syntax is incorrect. (type was *os.PathError)"

**Comment by mikeparker** [comment]
I think there are a few missing repro instructions, can I check these are correct:
- You've previously created the TestCreateDirectory folder on your C drive
- You've created a .net core test project targeting 3.1
- You've copied the test dll and any related artifacts into the TestCreateDirectory

Is this correct? Anything I've missed?

**Comment by mikeparker** [comment]
Ok, I have reproduced it. It turns out `dotnet test` creates a directory with `:` in, which is actually a duplicate of https://github.com/docker/for-win/issues/5520, e.g. `TestResults/Deploy_ 2020-01-22 14:14:56`

**Comment by mikeparker** [comment]
As a workaround it looks like using nUnit instead of MSTest works, as it outputs test results differently. Probably other test providers will work, or if you are tied to MSTest you can probably turn off the TestResults output or if you need it you may be able to configure the TestResults output folder name format differently.

**Comment by tronsoft** [comment]
Hi Mike,
This is correct. TestCreateDirectory is the root directory of my DotNet core 3.1 solution. It contains the solution file and a console app folder and a test project folder. The directory is mounted into the container on /app. I built the project and then I executed the unit test assembly with dotnet test ./bin/Debug/dotnetcoreapp3.1/myunittests.dll
I hope this explains my workflow better.
Kind regards,
Ton de Ron
Get Outlook for Android
